aries_vcx/src/handlers/issuance/issuer/state_machine.rs

```diff
@@ -369,23 +372,28 @@ pub mod test {
     use crate::messages::issuance::credential::test_utils::_credential;
     use crate::messages::issuance::credential_offer::test_utils::_credential_offer;
     use crate::messages::issuance::credential_proposal::test_utils::_credential_proposal;
-    use crate::messages::issuance::credential_request::test_utils::_credential_request;
+    use crate::messages::issuance::credential_request::test_utils::{_credential_request, _credential_request_1};
     use crate::messages::issuance::test::{_ack, _problem_report};
+    use crate::messages::connection::did_doc::DidDoc;
+    use crate::messages::a2a::A2AMessage;
+    use crate::messages::basic_message::message::BasicMessage;
     use crate::test::source_id;
+    use crate::utils::send_message;
     use crate::utils::devsetup::SetupMocks;
+    use agency_client::mocking::HttpClientMockResponse;
 
     use super::*;
 
-    fn _rev_reg_id() -> String {
+    pub fn _rev_reg_id() -> String {
         String::from("TEST_REV_REG_ID")
     }
 
-    fn _tails_file() -> String {
+    pub fn _tails_file() -> String {
         String::from("TEST_TAILS_FILE")
     }
 
-    fn _send_message() -> Option<&'static impl Fn(&A2AMessage) -> VcxResult<()>> {
-        Some(&|_: &A2AMessage| VcxResult::Ok(()))
+    pub fn _send_message() -> Option<&'static impl Fn(&A2AMessage) -> VcxResult<()>> {
+        Some(&|_: &A2AMessage| send_message("", &DidDoc::default(), &A2AMessage::BasicMessage(BasicMessage::default())))
```

Comment:
I am surprised tests are not failing after this change - this will actually try to call networking code. Why making this change?

Reply:
I think it would be cleaner to just make new method, something like `_send_message_but_fail` and return `Error`. 
Reasons:
1. It would be easier to read the test, imo
2. This adds dependency on agency code into state machine - I know such dependencies exists today, but we should head toward reducing them

Reply:
I like the flexibility of mocking responses, but I did as you suggested.